File: Miscellaneous/Golang/136. Single Number.go
package main

import "fmt"

func main() {

	arr := []int{4, 2, 1, 2, 1}
	result := singleNumber(arr)
	fmt.Println(result)
}

func singleNumber(arr []int) int {
	ans := arr[0]

	for i := 1; i < len(arr); i++ {
		ans ^= arr[i]
	}

	return ans
}

File: Miscellaneous/Python/136. Single Number.py
class Solution:
    def singleNumber(self, nums):
        ans = nums[0]

        for i in range(1, len(nums)):
            ans = ans ^ nums[i]

        return ans

obj = Solution()
arr = [2,2,1]
ans = obj.singleNumber(arr)
print(ans)